hollow/src/main/java/com/netflix/hollow/core/HollowConstants.java

```diff
@@ -35,4 +35,9 @@
      * An ordinal of NULL_ORDINAL signifies "null reference" or "no ordinal"
      */
     int ORDINAL_NONE = -1;
+
+    /**
+     * The maximum size of Hollow hash table; largest positive power of 2 that fits in an int
+     */
+    int HASH_TABLE_MAX_SIZE = 1 << 30;
```

Comment:
This should be the maximum number of buckets, not the maximum size.  Some empty space should be left over, otherwise performance will degrade significantly (approaching O(n) insertion and lookup).  The default selected was 70% populated, which seems to work well -- I would recommend a maximum size of `(1L << 30) * 7 / 10`. 

As a bonus, selecting this value would allow you to omit the overflow check -- values which would otherwise overflow would instead now throw your `IllegalArgumentException`